src/runtime_src/tools/scripts/_xbreset.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python3
+"""
+ Copyright (C) 2020 Xilinx, Inc
+ Author(s): Brian Xu (brianx@xlinx.com)
+
+ Licensed under the Apache License, Version 2.0 (the "License"). You may
+ not use this file except in compliance with the License. A copy of the
+ License is located at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ License for the specific language governing permissions and limitations
+ under the License.
+"""
+
+import sys
+import argparse
+import textwrap
+import os
+import re
+import subprocess
+import time
+
+node_bus_mapping = {}
+node_parent_mapping = {}
+bus_children = {}
+rootDir = "/sys/bus/pci/devices/"
+
+def get_node_bus_mapping():
+    subdir = os.listdir(rootDir)
```

Comment:
Please do add some comment regarding this flow.   This should include, but is not limited to:

1. An example of the path being parse
2.  Key flow points.  For example:
   a. Examine all of the directories looking for the "vendor" value "0x10ee"

Reply:
ok. will do